sssd_test_framework/roles/keycloak.py

```diff
@@ -0,0 +1,299 @@
+        :type member: list[KeycloakUser | KeycloakGroup]
+        :return: Self.
+        :rtype: KeycloakGroup
+        """
+        self._exec('remove-member', self.__get_member_args(members))
+        return self
+
+    def __get_member_args(self, members: list[KeycloakUser | KeycloakGroup]) -> list[str]:
+        users = [x for item in members if isinstance(item, KeycloakUser) for x in (f'users/{item.id}',)]
+        groups = [x for item in members if isinstance(item, KeycloakGroup) for x in (f'groups/{item.id}',)]
+        return [*users, *groups]
+
+
+class KeycloakDomain(object):
+    """
+    Keycloak integration domain management.
+    """
+
+    def __init__(self, role: Keycloak, name: str) -> None:
+        """
+        :param role: Keycloak role object.
+        :type role: Keycloak
+        :param name: Integration Domain name.
+        :type name: str
+        """
+        self.role = role
+        self.name = name
+        self.id: str = ""
+
+    def add(
+        self,
+        *,
+        scim_host: str,
+        scim_admin: str | None = "scim",
+        scim_password: str | None = "Secret123",
+        dom_name: str,
+        dom_url: str,
+        client_id: str | None = "admin",
+        client_secret: str | None = "Secret123",
+        id_provider: str | None = "ipa",
+        ldap_tls_cacert: str | None = None,
+    ) -> KeycloakDomain:
+        """
+        Create new Keycloak integration domain.
+
+        :param scim_host: SCIM Server Hostname
+        :type scim_host: str
+        :return: Self.
+        :rtype: KeycloakGroup
+        """
+
+        # run command to add group
+        create_domain = f"create components -r master -s name={self.name} " \
+                        f"-s providerId=scim " \
+                        f"-s providerType=org.keycloak.storage.UserStorageProvider " \
+                        f"-s config.scimurl=[\"{scim_host}:8000\"] " \
+                        f"-s config.loginusername=[\"{scim_admin}\"] " \
+                        f"-s config.loginpassword=[\"{scim_password}\"] " \
+                        f"-s config.domain=[\"{dom_url}\"] " \
+                        f"-s config.domainname=[\"{dom_name}\"] " \
+                        f"-s config.domaindesc=[\"Bridge_to_{id_provider}\"] " \
+                        f"-s config.domainurl=[\"https://{scim_host}\"] " \
+                        f"-s config.domainclientid=[\"{client_id}\"] " \
+                        f"-s config.domainclientsecret=[\"{client_secret}\"] " \
+                        f"-s config.idprovider=[\"{id_provider}\"] " \
+                        f"-s config.cacert=[\"{ldap_tls_cacert}\"] " \
+                        f"-s config.addintgdomain=[\"True\"] " \
+                        f"-s config.delintgdomain=[\"True\"]"
+
+        result = self.role.kcadm(create_domain)
+
+        return result.stdout
+
+    def ipa_fix(self, client: Client) -> None:
+        user_attrs = "mail:mail, sn:sn, givenname:givenname"
```

Comment:
Doc string, I don't know why this function is necessary, can you please add it to the description why this is needed?